tfe/resource_tfe_workspace.go

```diff
@@ -767,3 +805,12 @@ func resourceTFEWorkspaceImporter(ctx context.Context, d *schema.ResourceData, m
 func warnWorkspaceTagsCasing(ctx context.Context, tag string) {
 	log.Printf("[WARN] The tag \"%s\" contains uppercase characters that will be transformed by the API. Please update your configuration to lowercase tags in order to avoid conflicts with state.", tag)
 }
+
+func errWorkspaceSafeDeleteWithPermission(workspaceID string, err error) error {
+	if err != nil {
+		if strings.HasPrefix(err.Error(), "conflict") {
+			return fmt.Errorf("Error deleting workspace %s: %w\nTo delete this workspace without destroying the managed resources, add force_delete = true to the resource config.", workspaceID, err)
+		}
+	}
+	return nil
+}
```

Comment:
the logic here seems wrong; if there's an error, and it's not a conflict, this function should still return the `err` value. i saw this behavior today when i destroyed a terraform cloud workspace, and the workspace wasn't actually deleted but the apply succeeded and the resource was removed from state.